lib/src/visitor/async_evaluate.dart

```diff
@@ -199,11 +206,16 @@ class _EvaluateVisitor
   /// imports, it contains the URL passed to the `@import`.
   final _includedFiles = <String>{};
 
-  /// The set of canonical URLs for modules (or imported files) that are
-  /// currently being evaluated.
+  /// A map from canonical URLs for modules (or imported files) that are
+  /// currently being evaluated to AST nodes whose spans indicate the original
+  /// loads for those modules.
+  ///
+  /// Map values may be `null`, which indicates an active module that doesn't
+  /// have a source span associated with its original load (such as the
+  /// entrypoint module).
   ///
   /// This is used to ensure that we don't get into an infinite load loop.
-  final _activeModules = <Uri>{};
+  final _activeModules = <Uri, AstNode>{};
```

Comment:
nit: For consistency of naming with the other maps in this change I think this should be `_activeModuleNodes`.

Reply:
This is somewhat different than the other maps because it's not *just* used to track the nodes—whether the map has a given key also indicates which modules are active. So I think naming that could be misleading.